Sources/SwiftParser/Declarations.swift

```diff
@@ -211,15 +211,22 @@ extension Parser {
       break
     }
 
+    // If we are inside a memberDecl list, we don't want to eat closing braces (which most likely close the outer context)
+    // while recoverying to the declaration start.
+    let recoveryPrecedence = inMemberDeclList ? TokenPrecedence.closingBrace : nil
+
+    switch self.canRecoverTo(anyIn: DeclarationStart.self, overrideRecoveryPrecedence: recoveryPrecedence) {
+    case (.operatorKeyword, _)?:
+      return RawDeclSyntax(parseOperatorDeclaration())
+    default:
+      break
+    }
```

Comment:
You should be able to simplify this to

```suggestion
    if self.canRecoverTo(TokenSpec(.operator, recoveryPrecedence: recoveryPrecedence) != nil {
      return RawDeclSyntax(parseOperatorDeclaration())
    }
```

Reply:
If we only try to recover to ``.operator``, there will be some corner case e.g.
```swift
enum foo {
  case operator
}
```
where ``case`` is with ``stmtKeyword`` precedence and ``operator`` is with ``declKeyword`` precedence, which leads to some broken diagnostic messages.

On the other hand, I think have a switch control flow here can help us extend to those declarations that do not need ``DeclAttr`` later, such as ``precedenceGroupDecl``. WDYT?

Reply:
That makes sense to me. You have convinced me. 👍🏽